PowerVCF.psm1

```diff
@@ -2980,6 +2980,253 @@ Export-ModuleMember -Function Restart-CloudBuilderSDDCValidation
 
 
 
+######### Start APIs for managing SOS ##########
+
+Function Get-VCFHealthSummaryTask {
+    <#
+        .SYNOPSIS
+        Get Health Summary tasks
+
+        .DESCRIPTION
+        The Get-VCFHealthSummaryTask cmdlet retrieves the Health Summary tasks
+
+        .EXAMPLE
+        Get-VCFHealthSummaryTask
+        This example gets all Health Summary tasks
+
+        .EXAMPLE
+        Get-VCFHealthSummaryTask -id <task_id>
+        This example gets the Health Summary task by id
+    #>
+
+    Param (
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$id
+    )
+
+    Try {
+        $vcfVersion = ((Get-VCFManager).version -Split ('\.\d{1}\-\d{8}')) -split '\s+' -match '\S'
+        if ($vcfVersion -ge "4.4.0") {
+            createHeader # Calls createHeader function to set Accept & Authorization
+            checkVCFToken # Calls the CheckVCFToken function to validate the access token and refresh if necessary
+            if ($PsBoundParameters.ContainsKey("id")) {
+                $uri = "https://$sddcManager/v1/system/health-summary/$id"
+                $response = Invoke-RestMethod -Method GET -URI $uri -headers $headers
+                $response
+            }
+            else {
+                $uri = "https://$sddcManager/v1/system/health-summary"
+                $response = Invoke-RestMethod -Method GET -URI $uri -headers $headers
+                $response.elements
+            }
+        }
+        else { 
+            Write-Warning "API is only valid with VMware Cloud Foundation v4.4.0 or later"
+        }
+    }
+    Catch {
+        ResponseException -object $_
+    }
+}
+Export-ModuleMember -Function Get-VCFHealthSummaryTask
+
+Function Request-VCFHealthSummaryBundle {
+    <#
+        .SYNOPSIS
+        Download Health Summary bundle
+
+        .DESCRIPTION
+        The Request-VCFHealthSummaryBundle cmdlet downloads the Health Summary bundle
+
+        .EXAMPLE
+        Request-VCFHealthSummaryBundle -id <task_id>
+        This example downloads a Health Summary bundle
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$id
+    )
+
+    Try {
+        $vcfVersion = ((Get-VCFManager).version -Split ('\.\d{1}\-\d{8}')) -split '\s+' -match '\S'
+        if ($vcfVersion -ge "4.4.0") {
+            checkVCFToken # Calls the CheckVCFToken function to validate the access token and refresh if necessary
+            $vcfHeaders = @{"Accept" = "application/octet-stream" }
+            $vcfHeaders.Add("ContentType", "application/octet-stream")
+            $vcfHeaders.Add("Authorization", "Bearer $accessToken")
+            $uri = "https://$sddcManager/v1/system/health-summary/$id/data"
+            Invoke-RestMethod -Method GET -URI $uri -headers $vcfHeaders -OutFile "health-summary-$id.tar"
```

Comment:
This will output the file to the current working directory? Maybe a future enhancement to allow specifying the target dir?